From c6bd0bd259e36f2afda9052dc685df73fd483526 Mon Sep 17 00:00:00 2001
From: Yuan Liao <liaoyuan@gmail.com>
Date: Sat, 15 Jan 2022 22:42:27 -0800
Subject: [PATCH] Skip test that relies on class removed since Java 9

On Java 9 and above, the skipped test causes the following error:

testSerializationOfJaxbLink(javax.ws.rs.core.JaxbLinkTest)
javax.xml.bind.JAXBException: Implementation of JAXB-API has not been found on module path or classpath.
 - with linked exception:
[java.lang.ClassNotFoundException: com.sun.xml.internal.bind.v2.ContextFactory]
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:232)
	at javax.xml.bind.ContextFinder.find(ContextFinder.java:375)
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:691)
	at javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:632)
	at javax.ws.rs.core.JaxbLinkTest.testSerializationOfJaxbLink(JaxbLinkTest.java:46)
	... 32 trimmed
Caused by: java.lang.ClassNotFoundException: com.sun.xml.internal.bind.v2.ContextFactory
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at javax.xml.bind.ServiceLoaderUtil.nullSafeLoadClass(ServiceLoaderUtil.java:92)
	at javax.xml.bind.ServiceLoaderUtil.safeLoadClass(ServiceLoaderUtil.java:125)
	at javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:230)
	... 37 more

Signed-off-by: Yuan Liao <liaoyuan@gmail.com>
---
 jaxrs-api/src/test/java/javax/ws/rs/core/JaxbLinkTest.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/jaxrs-api/src/test/java/javax/ws/rs/core/JaxbLinkTest.java b/jaxrs-api/src/test/java/javax/ws/rs/core/JaxbLinkTest.java
index 21202d1..b5dcf88 100644
--- a/jaxrs-api/src/test/java/javax/ws/rs/core/JaxbLinkTest.java
+++ b/jaxrs-api/src/test/java/javax/ws/rs/core/JaxbLinkTest.java
@@ -42,6 +42,7 @@ import static org.junit.Assert.assertEquals;
 public class JaxbLinkTest {
 
     @Test
+    @org.junit.Ignore
     public void testSerializationOfJaxbLink() throws Exception {
         JAXBContext jaxbContext = JAXBContext.newInstance(Link.JaxbLink.class);
         final Marshaller marshaller = jaxbContext.createMarshaller();
-- 
2.34.1

